core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -1383,6 +1383,7 @@ private[spark] class DAGScheduler(
 
     event.reason match {
       case Success =>
+        taskScheduler.markPartitionCompletedFromEventLoop(task.partitionId, task.stageId)
```

Comment:
This call from thread  `dag-scheduler-event-loop` do not acquire a lock on `TaskScheduler`, so I think race condition still exists, since `handleFailedTask` is called from thread `task-result-getter`.

Reply:
The method `handleFailedTask` is called from TaskResultGetter to TaskSchedulerImpl which executes the call to TaskSetManager in synchronized block. So, my code in TaskSetManager runs within the TaskSchedulerImpl lock.

Reply:
@pgandhi999 I don't follow your explanation -- I think I agree w/ @Ngone51 .  yes, TSM calls `DAGs.taskEnded` while it has a lock on the TaskSchedulerImpl, but the `DAGs.taskEnded` call just puts an event into the queue and then returns.  So this part is happening w/ out the lock.  but maybe you meant something else?